Libraries/Opc.Ua.Client/Session.cs

```diff
@@ -3627,6 +3669,301 @@ public bool RemoveTransferredSubscription(Subscription subscription)
+            foreach (var error in errors)
+            {
+                previousErrors.Add(new ReferenceWrapper<ServiceResult> { reference = error });
+                errorAnchors.Add(previousErrors.Last());
+            }
+
+            byte[] nullByte = new byte[16];
+
+            var nextContinuationPoints = new ByteStringCollection();
+            var nextResults = new List<ReferenceDescriptionCollection>();
+            var nextErrors = new List<ReferenceWrapper<ServiceResult>>();            
+
+            for(int ii = 0; ii< nodeIds.Count; ii++)
+            {
+                if (continuationPoints[ii] != null && !ByteArraysEqual(continuationPoints[ii], nullByte))
+                {
+                    if (!StatusCode.IsBad(previousErrors[ii].reference.StatusCode))
+                    {
+                        nextContinuationPoints.Add(continuationPoints[ii]);
+                        nextResults.Add(previousResults[ii]);
+                        nextErrors.Add( previousErrors[ii] );
+                    }
+                    // ToDo: status code is bad and continuation point is not null
+                }
+            }
+
+            while(nextContinuationPoints.Count > 0)
+            { 
+
+                BrowseNext(null,
+                    false,
+                    nextContinuationPoints,
+                    out var revisedContinuationPoints,
+                    out var browseNextResults,
+                    out var browseNextErrors);
+
+                
+                for(int ii = 0; ii < browseNextResults.Count; ii++)
+                {
+                    nextResults[ii].AddRange(browseNextResults[ii]);
+                    nextErrors[ii].reference = browseNextErrors[ii] ;
+                }
+
+                previousResults = nextResults;
+                previousErrors =  nextErrors;
+
+                nextResults = new List<ReferenceDescriptionCollection>();
+                nextErrors  = new List<ReferenceWrapper<ServiceResult>>();
+                nextContinuationPoints = new ByteStringCollection();
+
+                for (int ii = 0; ii < revisedContinuationPoints.Count; ii++)
+                {
+                    if(revisedContinuationPoints[ii] != null && !ByteArraysEqual(revisedContinuationPoints[ii], nullByte))
+                    {
+                        nextContinuationPoints.Add(revisedContinuationPoints[ii]);
+                        nextResults.Add(previousResults[ii]);
+                        nextErrors.Add( previousErrors[ii] );
+                    }
+                }
+
+            }
+            finalErrors = new List<ServiceResult>();
+            foreach (var errorReference in errorAnchors)
+            {
+                finalErrors.Add(errorReference.reference);
+            }
+        }
+
+        // see https://stackoverflow.com/questions/43289/comparing-two-byte-arrays-in-net
+        private static bool ByteArraysEqual(ReadOnlySpan<byte> a1, ReadOnlySpan<byte> a2)
```

Comment:
I think Utils.Equal provides the same functionality

Reply:
Thanks! You are right of course. And even if that method would not already exist in Utils.cs such a method should go there. Fixed.